File: data/src/main/java/org/oppia/data/backends/gae/model/WrittenTranslation.java
package org.oppia.data.backends.gae.model;

import com.squareup.moshi.Json;

public class WrittenTranslation {

  /*
   * Parent Hierarchy: ExplorationContainer -> Exploration -> State -> WrittenTranslations -> WrittenTranslation
   */

  //TODO: HTML-String data type
  @Json(name = "html")
  public String html;

  @Json(name = "needs_update")
  public boolean needsUpdate;

}

File: data/src/main/java/org/oppia/data/backends/gae/model/WrittenTranslation.kt
package org.oppia.data.backends.gae.model

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
data class WrittenTranslation(

  @Json(name = "html") val html: String,
  @Json(name = "needs_update") val isUpdateNeeded: Boolean

)
